legal-api/src/legal_api/reports/report.py

```diff
@@ -315,6 +320,144 @@ def _format_incorporation_data(self, filing):
         else:
             filing['shareClasses'] = filing['incorporationApplication']['shareStructure']['shareClasses']
 
+    def _has_change(self, old_value, new_value):  # pylint: disable=no-self-use;
+        """Check to fix the hole in diff.
+
+        example:
+            old_value: None and new_value: ''
+            In reality there is no change but diff track it as a change
+        """
+        has_change = True  # assume that in all other cases diff has a valid change
+        if isinstance(old_value, str) and new_value is None:
+            has_change = old_value != ''
+        elif isinstance(new_value, str) and old_value is None:
+            has_change = new_value != ''
+
+        return has_change
+
+    def _format_with_diff_data(self, filing):
+        if incorporation_application := filing['incorporationApplication']:
+            diff = filing.get('correction', {}).get('diff', [])
+
+            self._format_name_translations_with_diff_data(filing, diff)
+            self._format_office_with_diff_data(incorporation_application, diff)
+            self._format_party_with_diff_data(incorporation_application, diff)
+            self._format_share_class_with_diff_data(incorporation_application, diff)
+
+    def _format_name_translations_with_diff_data(self, filing, diff):  # pylint: disable=no-self-use;
```

Comment:
This function need to be refactored after (Change Translated Names to managed array#5684)

Reply:
the change should be there to pick up.